qiskit/algorithms/evolvers/pvqd/pvqd.py

```diff
@@ -0,0 +1,364 @@
+                :math:`[-0.01, 0.01]`.
+
+        Returns:
+            A tuple consisting of the next parameters and the fidelity of the optimization.
+        """
+        loss, gradient = self.get_loss(hamiltonian, dt, theta)
+
+        if initial_guess is None:
+            initial_guess = np.random.random(self.initial_parameters.size) * 0.01
+
+        if isinstance(self.optimizer, Optimizer):
+            optimizer_result = self.optimizer.minimize(loss, initial_guess, gradient)
+        else:
+            optimizer_result = self.optimizer(loss, initial_guess, gradient)
+
+        # clip the fidelity to [0, 1]
+        fidelity = np.clip(1 - optimizer_result.fun, 0, 1)
+
+        return theta + optimizer_result.x, fidelity
+
+    def get_loss(
+        self,
+        hamiltonian: OperatorBase,
+        dt: float,
+        current_parameters: np.ndarray,
+    ) -> Tuple[Callable[[np.ndarray], float], Optional[Callable[[np.ndarray], np.ndarray]]]:
+
+        """Get a function to evaluate the infidelity between Trotter step and ansatz.
+
+        Args:
+            hamiltonian: The Hamiltonian under which to evolve.
+            dt: The time step.
+            current_parameters: The current parameters.
+
+        Returns:
+            A callable to evaluate the infidelity and, if gradients are supported and required,
+                a second callable to evaluate the gradient of the infidelity.
+        """
+        self._validate_setup()
+
+        # use Trotterization to evolve the current state
+        trotterized = self.ansatz.bind_parameters(current_parameters)
+        trotterized.append(
+            PauliEvolutionGate(hamiltonian, time=dt, synthesis=self.evolution), self.ansatz.qubits
+        )
+
+        # define the overlap of the Trotterized state and the ansatz
+        x = ParameterVector("w", self.ansatz.num_parameters)
+        shifted = self.ansatz.assign_parameters(current_parameters + x)
+        overlap = StateFn(trotterized).adjoint() @ StateFn(shifted)
+
+        converted = self.expectation.convert(overlap)
+
+        def evaluate_loss(
+            displacement: Union[np.ndarray, List[np.ndarray]]
+        ) -> Union[float, List[float]]:
+            """Evaluate the overlap of the ansatz with the Trotterized evolution.
+
+            Args:
+                displacement: The parameters for the ansatz.
+
+            Returns:
+                The fidelity of the ansatz with parameters ``theta`` and the Trotterized evolution.
+            """
+            if isinstance(displacement, list):
+                displacement = np.asarray(displacement)
+                value_dict = {x_i: displacement[:, i].tolist() for i, x_i in enumerate(x)}
+            else:
+                value_dict = dict(zip(x, displacement))
+
+            sampled = self._sampler.convert(converted, params=value_dict)
+            return 1 - np.abs(sampled.eval()) ** 2
```

Comment:
Same as above. Do we maybe (in the future) enable custom loss functions?

Reply:
That's a good idea! I'll add a comment but since the paper only uses this loss I would prefer to not add a feature we might not need -- but for sure it's easy to add if we want to 🙂